stabilizer/src/main/java/com/hazelcast/stabilizer/tests/map/PredicatesTest.java

```diff
@@ -0,0 +1,438 @@
+
+    @Run
+    public void run() {
+        ThreadSpawner spawner = new ThreadSpawner(testContext.getTestId());
+        for (int k = 0; k < threadCount; k++) {
+            spawner.spawn(new Worker());
+        }
+        spawner.awaitCompletion();
+    }
+
+    @Performance
+    public long getOperationCount() {
+        return operations.get();
+    }
+
+    private class Worker implements Runnable {
+        @Override
+        public void run() {
+            long iteration = 0;
+            Metronome metronome = SimpleMetronome.withFixedIntervalMs(intervalMs);
+            SqlPredicate sqlPredicate = new SqlPredicate(sqlQuery);
+            PredicateType type = selector.select();
+            AgePredicate agePredicate = new AgePredicate(30);
+            NamePredicate namePredicate = new NamePredicate("aaa");
+            ActivePredicate activePredicate = new ActivePredicate(true);
+            SalaryPredicate salaryPredicate = new SalaryPredicate(700.0);
+            while (!testContext.isStopped()) {
+                metronome.waitForNext();
+                search.started();
+                if (customPredicate) {
+                    switch (type){
+                        case AGE:
+                            map.values(agePredicate);
+                            break;
+                        case NAME:
+                            map.values(namePredicate);
+                            break;
+                        case ACTIVE:
+                            map.values(activePredicate);
+                            break;
+                        case SALARY:
+                            map.values(salaryPredicate);
+                            break;
+                        default:
+                            log.severe("There is no predicate type like this");
+                    }
+                } else {
+                    map.values(sqlPredicate);
+                }
+                search.done();
+
+                iteration++;
+                if (iteration % logFrequency == 0) {
+                    log.info(Thread.currentThread().getName() + " At iteration: " + iteration);
+                }
+
+                if (iteration % performanceUpdateFrequency == 0) {
+                    operations.addAndGet(performanceUpdateFrequency);
+                }
+
+            }
+            operations.addAndGet(iteration % performanceUpdateFrequency);
+        }
+    }
+
+    public EmployeeImpl createEmployee(int id) {
+        switch (serializationStrategyWithNumber) {
+            case 1:
+                return new DataSerializableEmployee(id);
+            case 2:
+                return new SerializableEmployee(id);
+            case 3:
+                return new IdentifiedDataSerializableEmployee(id);
+            case 4:
+                return new PortableEmployee(id);
+        }
+        return null;
+    }
+
+    //Custom Predicates
+    private static class NamePredicate implements Predicate<String, EmployeeImpl> {
```

Comment:
I think we can get rid of all these predicates since they are all of the same complexity.